packages/authentication/test/unit/strategy-adapter.test.ts

```diff
@@ -0,0 +1,65 @@
+// Copyright IBM Corp. 2013,2017. All Rights Reserved.
+// Node module: loopback
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect} from 'testlab';
+import {ShimRequest} from '../..';
+import {StrategyAdapter} from '../..';
+import {ParsedRequest} from '../..';
+const passportStrategy = require('passport-strategy');
+
+describe('Strategy Adapter', () => {
+  const mockUser: User = {id: 'mock-user', role: 'mock-role'};
+  interface User {
+    id: string;
+    role: string;
+  }
+
+  describe('authenticate()', () => {
+
+    it('calls the authenticate method of the strategy', (done) => {
+      const strategy = new passportStrategy();
+      let calledFlag = false;
+      strategy.authenticate = function() {
+        calledFlag = true;
+      };
+      const adapter = new StrategyAdapter(strategy);
+      const request = new ShimRequest();
+      // tslint:disable-next-line:no-floating-promises
+      adapter.authenticate(request as ParsedRequest);
+      expect(calledFlag).to.be.true();
+      done();
+    });
+
+    it('adds success handler to the strategy', (done) => {
+      const strategy = new MockStrategy();
+      const adapter = new StrategyAdapter(strategy);
+      const request = new ShimRequest();
+      // tslint:disable-next-line:no-floating-promises
+      adapter.authenticate(request as ParsedRequest)
```

Comment:
Please don't mix callback-based style `(done) => ` with promise-based APIs. Mocha supports promises out of the box, just return the final promise from your test function. You can use `async/await` too!

```ts
it('adds success handler to the strategy', async () => {
  // ...
  const user = await adapter.authenticate(request);
  expect(user).to.be.eql(mockUser);
});
```

Also why are we explicitly casting `ShimRequest` to `ParsedRequest`? If `ShimRequest` does not provide all properties described in `ParsedRequest`, then the code relying on `ParsedRequest` may break at runtime. Which is the thing we are trying to avoid by using TypeScript, where such errors can be caught at compile time.